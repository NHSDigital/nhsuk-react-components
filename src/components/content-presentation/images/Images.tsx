import React, { FC, HTMLProps } from 'react';
import classNames from 'classnames';

interface ImageProps extends HTMLProps<HTMLImageElement> {
  // Overriding the default crossOrigin the default is crossOrigin: string | undefined
  // which causes a typescript "incompatible types" error.
  crossOrigin?: 'anonymous' | 'use-credentials' | undefined;
  caption?: string;
}

const Images: FC<ImageProps> = ({ className, caption, ...rest }) => (
  <figure className="nhsuk-image">
    {/* eslint-disable-next-line jsx-a11y/alt-text */}
    <img className={classNames('nhsuk-image__img', className)} {...rest} />
    {caption ? <figcaption className="nhsuk-image__caption">{caption}</figcaption> : null}
  </figure>
);

export default Images;
