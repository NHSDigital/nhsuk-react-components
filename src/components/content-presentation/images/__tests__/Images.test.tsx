import React from 'react';
import { shallow } from 'enzyme';
import Images from '..';

describe('Images', () => {
  it('matches snapshot', () => {
    const component = shallow(<Images />);
    expect(component.hasClass('nhsuk-image')).toBeTruthy();
    expect(component).toMatchSnapshot('Images');
    component.unmount();
  });

  it('renders caption', () => {
    const component = shallow(<Images caption="Caption" />);
    expect(component.find('figcaption').exists()).toBeTruthy();
    expect(component.find('figcaption').hasClass('nhsuk-image__caption')).toBeTruthy();
    expect(component.find('figcaption').text()).toBe('Caption');
    component.unmount();
  });
});
