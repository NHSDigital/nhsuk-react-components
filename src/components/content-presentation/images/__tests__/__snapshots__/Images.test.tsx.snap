// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Images matches snapshot: Images 1`] = `
<figure
  className="nhsuk-image"
>
  <img
    className="nhsuk-image__img"
  />
</figure>
`;
